### PR TITLE
Unsteady Aero Driver:  adding 3 degrees of freedom for the motion of a genralized airfoil section

### PR DESCRIPTION
**Pull request by ebranlard**
This pull request is soon ready to be merged

**Feature or improvement description**
Adding 3 degrees of freedom fro the motion of a genralized airfoil section 

**Impacted areas of the software**
Unsteady Aerodynamics 

**Additional supporting information**
A paper will be released at the torque 2024 conference. A link is provided in the documentation. 

**Test results, if applicable**
Input file is backward compatible

- [x] Update the documentation  - Live version [here](https://ebranlard-openfast.readthedocs.io/en/f-ua-elast-dev/source/user/aerodyn/theory_ua.html#driver) 
- [x] Update r-test input files
- [x] Submit a new r-test for aeroelastic simulation will be done later after torque paper
- [ ] Quick review

Example input file:
[UA2.txt](https://github.com/OpenFAST/openfast/files/13519950/UA2.txt)

### CODE REVIEW

**Comment by andrew-platt** [comment]
I see the `LynDyn` module is new.  Is that only used by the unsteady aero?

**Comment by bjonkman** [comment]
> I see the `LynDyn` module is new. Is that only used by the unsteady aero?

Yes. That is only used in the Unsteady Aero driver.

File: modules/openfast-library/CMakeLists.txt [hunk 1]
```diff
@@ -55,6 +55,7 @@ target_link_libraries(openfast_prelib
   icedynlib
   icefloelib
   ifwlib
+  lindynlib
```
Comment (andrew-platt) [inline modules/openfast-library/CMakeLists.txt:58]: If `LynDyn` is only used in the UA driver, we don't need to include it in the openfast-library.  I noticed it is not included in the CMakeLists.txt for Simulink or in the FAST VS projects.

Comment (andrew-platt) [reply modules/openfast-library/CMakeLists.txt]: resolved with 1dd60b96d

**Pull request merged**